bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableDataGrpcClient.java

```diff
@@ -79,53 +78,62 @@
 
   @Mock ClientCall mockClientCall;
 
-  @Mock ScheduledExecutorService mochScheduler;
+  @Mock ScheduledExecutorService mockScheduler;
 
-  BigtableDataGrpcClient defaultClient;
-
-  @Before
-  public void setup() {
-    when(mockChannel.newCall(any(MethodDescriptor.class), any(CallOptions.class)))
-        .thenReturn(mockClientCall);
-    defaultClient = createClient(false);
+  private BigtableDataGrpcClient createClient(boolean shouldChannelCallMocked) {
+    return createClient(false, true, shouldChannelCallMocked);
   }
 
-  protected BigtableDataGrpcClient createClient(boolean allowRetriesWithoutTimestamp) {
+  private BigtableDataGrpcClient createClient(
+      boolean allowRetriesWithoutTimestamp,
+      boolean shouldChannelCallMocked,
+      boolean shouldClientCallMocked) {
+
     RetryOptions retryOptions =
         RetryOptions.builder()
             .setAllowRetriesWithoutTimestamp(allowRetriesWithoutTimestamp)
             .build();
     BigtableOptions options = BigtableOptions.builder().setRetryOptions(retryOptions).build();
-    doAnswer(
-            new Answer<Void>() {
-              @Override
-              public Void answer(InvocationOnMock invocation) throws Throwable {
-                checkHeader(invocation.getArgument(1, Metadata.class));
-                return null;
-              }
-            })
-        .when(mockClientCall)
-        .start(any(ClientCall.Listener.class), any(Metadata.class));
-    return new BigtableDataGrpcClient(mockChannel, mochScheduler, options);
+    if (shouldChannelCallMocked) {
+      when(mockChannel.newCall(any(MethodDescriptor.class), any(CallOptions.class)))
+          .thenReturn(mockClientCall);
+    }
+    if (shouldClientCallMocked) {
+      doAnswer(
+              new Answer<Void>() {
+                @Override
+                public Void answer(InvocationOnMock invocation) {
+                  checkHeader(invocation.getArgument(1, Metadata.class));
+                  return null;
+                }
+              })
+          .when(mockClientCall)
+          .start(any(ClientCall.Listener.class), any(Metadata.class));
+    }
+
+    return new BigtableDataGrpcClient(mockChannel, mockScheduler, options);
```

Comment:
This makes the actual tests extremely hard to read. 
Reading the test code I have no idea what this does: `createClient(false, false, false);` and I have scroll 

Lets put this on hold until we finish the veneer migration